iOS/MobileStarter/react-app/src/IModelDownloader.tsx

```diff
@@ -0,0 +1,115 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import React from "react";
+import { useIsMountedRef } from "@itwin/mobile-ui-react";
+import { Project } from "@itwin/projects-client";
+import { ProgressInfo } from "@itwin/core-frontend/lib/cjs/request/Request";
+import { DownloadBriefcaseOptions, NativeApp } from "@itwin/core-frontend";
+import { MinimalIModel } from "@itwin/imodels-client-management";
+import { BentleyError, BriefcaseDownloader, BriefcaseStatus, IModelStatus, SyncMode } from "@itwin/core-common";
+import { ProgressRadial } from "@itwin/itwinui-react";
+import { Button, i18n, presentError, IModelInfo } from "./Exports";
+
+async function downloadIModel(project: Project, iModel: MinimalIModel, handleProgress: (progress: ProgressInfo) => boolean): Promise<IModelInfo> {
```

Comment:
Looking at how this code is implemented, and how it is used, I feel that this should return Promise<IModelInfo | undefined>, and return undefined when it gets canceled. The minimalIModel inside the return doesn't do anything useful in the cancel case, so I feel that that field should be required, but the whole IModelInfo itself should be optional.

Reply:
I agree the signature needs improvement, but feel it should return Promise<LocalBriefcaseProps | undefined>. Do you agree?

Reply:
Since MinimalIModel is an input param, yes.

Reply:
Fixed